### PR TITLE
Implement hard-sigmoid

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented hard sigmoid function.

### CODE REVIEW

File: chainer/functions/activation/hard_sigmoid.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from chainer import function
+
+
+class HardSigmoid(function.Function):
+
+    """Hard-sigmoid funciton."""
+
+    def forward_cpu(self, inputs):
+        x = inputs[0]
+        return numpy.minimum(1.0, numpy.maximum(0.0, x * 0.2 + 0.5)),
```
Comment (okuta) [inline chainer/functions/activation/hard_sigmoid.py]: How about using `numpy.clip`?

File: tests/chainer_tests/functions_tests/activation_tests/test_hard_sigmoid.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    shape = (3, 4)
+
+    def setUp(self):
+        self.x = numpy.random.uniform(-1, 1, self.shape).astype(numpy.float32)
+        self.g = numpy.random.uniform(-1, 1, self.shape).astype(numpy.float32)
+
+    def check_forward(self, x_data):
+        x = chainer.Variable(x_data)
+        y = functions.hard_sigmoid(x)
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/activation_tests/test_hard_sigmoid.py]: Please check `y.data.dtype`.

File: tests/chainer_tests/functions_tests/activation_tests/test_hard_sigmoid.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+from chainer.testing import condition
+
+
+class TestHardSigmoid(unittest.TestCase):
+
+    shape = (3, 4)
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/activation_tests/test_hard_sigmoid.py]: Please check `shape = ()`.

**Comment by okuta** [comment]
Please add `check_type_forward`.

**Comment by okuta** [comment]
LGTM except comments.

**Comment by unnonouno** [comment]
I updated it.

**Comment by okuta** [comment]
Thank you for adding all floating point test.

File: tests/chainer_tests/functions_tests/activation_tests/test_hard_sigmoid.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+from chainer.testing import condition
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(3, 4), ()],
+    'dtype': [numpy.float16, numpy.float32, numpy.float64]
+}))
+class TestHardSigmoid(unittest.TestCase):
+
+    shape = (3, 4)
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/activation_tests/test_hard_sigmoid.py]: Please remove this line.

**Comment by unnonouno** [comment]
I fixed it.

**Pull request merged**